openlibrary/plugins/openlibrary/js/readmore.js

```diff
@@ -52,10 +52,13 @@ export class ReadMoreComponent {
     }
 
     reset() {
-        this.fullHeight = this.$content.scrollHeight;
+        const fullHeight = this.$content.scrollHeight;
+        const collapsedHeight = this.collapsedHeight;
+        const readMoreButtonHeight = this.$readMoreButton ? this.$readMoreButton.offsetHeight : 0;
```

Comment:
There's a little gotcha here with the `$readMoreButton` height. If the readMoreButton might be hidden when this method is called, the offsetHeight will be zero, even though the button will still have a height once it's made visible! The way we can handle this, is by recording the height of the readMoreButton in the constructor of this class, and updating the height in `reset` if it's non-zero. The readMoreButton is visible, by default when the class is first constructed, so at that point in time, it should have a non-zero height. And then here we can update it.

E.g.

```js
// constructor
this.readMoreHeight = this.$readMoreButton.offsetHeight;

// reset
if (this.$readMoreButton.offsetHeight) {
    this.readMoreHeight = this.$readMoreButton.offsetHeight;
}
```

I think that should handle everything correctly!